src/main/java/org/opensearch/agent/tools/PPLTool.java

```diff
@@ -189,7 +197,78 @@ public <T> void run(Map<String, String> parameters, ActionListener<T> listener)
+            ActionRequest request = new MLPredictionTaskRequest(
+                modelId,
+                MLInput.builder().algorithm(FunctionName.REMOTE).inputDataset(inputDataSet).build(),
+                null
+            );
+            client.execute(MLPredictionTaskAction.INSTANCE, request, ActionListener.wrap(mlTaskResponse -> {
+                ModelTensorOutput modelTensorOutput = (ModelTensorOutput) mlTaskResponse.getOutput();
+                ModelTensors modelTensors = modelTensorOutput.getMlModelOutputs().get(0);
+                ModelTensor modelTensor = modelTensors.getMlModelTensors().get(0);
+                Map<String, String> dataAsMap = (Map<String, String>) modelTensor.getDataAsMap();
+                if (dataAsMap.get("response") == null) {
+                    listener.onFailure(new IllegalStateException("Remote endpoint fails to inference."));
+                    return;
+                }
+                String ppl = parseOutput(dataAsMap.get("response"), indexName);
+                if (!this.execute) {
+                    Map<String, String> ret = ImmutableMap.of("ppl", ppl);
+                    listener.onResponse((T) AccessController.doPrivileged((PrivilegedExceptionAction<String>) () -> gson.toJson(ret)));
+                    return;
+                }
+                JSONObject jsonContent = new JSONObject(ImmutableMap.of("query", ppl));
+                PPLQueryRequest pplQueryRequest = new PPLQueryRequest(ppl, jsonContent, null, "jdbc");
+                TransportPPLQueryRequest transportPPLQueryRequest = new TransportPPLQueryRequest(pplQueryRequest);
+                client
+                    .execute(
+                        PPLQueryAction.INSTANCE,
+                        transportPPLQueryRequest,
+                        getPPLTransportActionListener(ActionListener.wrap(transportPPLQueryResponse -> {
+                            String results = transportPPLQueryResponse.getResult();
+                            Map<String, String> returnResults = ImmutableMap.of("ppl", ppl, "executionResult", results);
+                            listener
+                                .onResponse(
+                                    (T) AccessController.doPrivileged((PrivilegedExceptionAction<String>) () -> gson.toJson(returnResults))
+                                );
+                        }, e -> {
+                            String pplError = "execute ppl:" + ppl + ", get error: " + e.getMessage();
+                            Exception exception = new Exception(pplError);
+                            listener.onFailure(exception);
+                        }))
+                    );
+                // Execute output here
+            }, e -> {
+                log.error(String.format(Locale.ROOT, "fail to predict model: %s with error: %s", modelId, e.getMessage()), e);
+                listener.onFailure(e);
+            }));
+        }, e -> {
+            log.info("fail to get table info");
```

Comment:
Table info here stands for `index schema`, right? If so, it's better to change it to `index schema` so that user won't confused.